runtime/src/accounts.rs

```diff
@@ -931,6 +931,26 @@ impl Accounts {
             .collect()
     }
 
+    #[allow(clippy::needless_collect)]
+    pub fn lock_accounts_with_results<'a>(
```

Comment:
Since we don't lock tx's that fail with cost tracker errors, we should include those errors in the `unlock_accounts` method `match` so that we don't unlock accounts that weren't locked.

Reply:
Did add additional qos related error code to `unlock_account` in #21406. This sounds bit loose, it is possible to have new error code that skips locking accounts, perhaps can refactor the lock/unlock accounts part by keeping locking states